connectors/mongo/conn.go

```diff
@@ -376,30 +395,44 @@ func (c *conn) ListData(ctx context.Context, r *connect.Request[adiomv1.ListData
 		}
 	}
 	c.buffersMutex.RLock()
-	ch, ok := c.buffers[cursorID]
+	buffer, ok := c.buffers[cursorID]
 	c.buffersMutex.RUnlock()
 	if !ok {
 		return nil, connect.NewError(connect.CodeInvalidArgument, fmt.Errorf("invalid cursor"))
 	}
+	if ctr+1 == buffer.ctr && buffer.last != nil { // Handle repeated requests for same page
+		buffer.lastUsed = time.Now()
+		buffer.cleanup.Reset(5 * time.Minute)
```

Comment:
this part same configuration and theres another one in the other place we call reset